mitiq/calibration/calibrator.py

```diff
@@ -92,208 +120,68 @@ def get_cost(self) -> Dict[str, int]:
             "ideal_executions": ideal,
         }
 
-    def run_circuits(self) -> List[Dict[str, Any]]:
-        """Runs all the circuits required for calibration.
-
-        Returns:
-            A collection of experimental results along with a summary of each
-            :class:`BenchmarkProblem` that was run.
-        """
-        expvals = []
+    def run(self) -> None:
+        """Runs all the circuits required for calibration."""
         for problem in self.circuits:
-            circuit, distribution = problem["circuit", "ideal_distribution"]
+            circuit = problem.circuit
             circuit.append(cirq.measure(circuit.all_qubits()))
```

Comment:
What about adding measurements in `make_problems()`, i.e. in the current `make_circuits()`?
Any disadvantage?

Reply:
Hmmm we totally can do that, but I think there are some minor tradeoffs.

1. [pro] The thing we are storing is the thing we are actually going to run, so no modifications are needed.
2. [pro] The user doesn't have to guess how circuits are being measured if they inspect them.
3. [con] Computing `two_qubit_gate_count` is slightly more cumbersome, and `depth` becomes somewhat ambiguous. E.g. does `-X-H-M-` have depth 2, or 3?

Aesthetically, I have a preference for storing the circuit without measurements, but that's only very minor. WDYT?

Reply:
Fine to keep measurements in `run()`